### PR TITLE
fix native toggle switch event in controlled mode

### LINKED ISSUE #2058: @axa-ch/toggle-switch controlled mode wrong checked state onChange
@axa-ch/toggle-switch version 1.1.2

When using toggle switch in controlled mode onChange callback will have wrong checked flag in event.target.

## Reproduction link
https://codesandbox.io/s/suspicious-bhaskara-tw809?file=/src/App.js

## Expected Behavior
onChange callback event is always in sync with UI display of toggle switch.

## Current Behavior
onChange callback event is independent of the UI display of toggle switch.

### PR DESCRIPTION
**Pull request by raphaellueckl**
Fixes #2058

# Merge Checklist:
- [x] Code selfreview has been performed ([Best Practices](https://github.com/axa-ch/patterns-library/blob/develop/CONTRIBUTION.md#best-practices)).
- [x] Tests are sufficient.
- [x] Modifications still work in IE.
- [x] Documentation is up to date.
- [x] Changelog is up to date.
- [x] Typescript typings for properties are up to date.
- [x] Designers approved changes or no approval needed.
- [x] Dependencies to other components still work.

### CODE REVIEW

File: src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+    this.state = {
+      active: true,
+      receivedEvent: '-',
+    };
+  }
+
+  change = ev => {
+    this.setState({ active: ev.target.checked });
+    this.setState({ receivedEvent: String(ev.target.checked) });
```
Comment (markus-walther) [inline src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx]: I think you can combine this into a single setState... (but see below, maybe active not even needed)

File: src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    super();
+
+    this.state = {
+      active: true,
+      receivedEvent: '-',
+    };
+  }
+
+  change = ev => {
+    this.setState({ active: ev.target.checked });
```
Comment (markus-walther) [inline src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx]: this is somewhat hard to read without a comment like // ev.target.checked expected to be constant true

File: src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    this.setState({ receivedEvent: String(ev.target.checked) });
+  };
+
+  render() {
+    return (
+      <div>
+        <AXAToggleSwitchReact
+          active={this.state.active}
```
Comment (markus-walther) [inline src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx]: No need to put this into state, wouldn't a constant be enough, like in the reproduction sandbox code? This here makes it more tricky to reason about things.

File: src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    };
+  }
+
+  change = ev => {
+    this.setState({ active: ev.target.checked });
+    this.setState({ receivedEvent: String(ev.target.checked) });
+  };
+
+  render() {
+    return (
```
Comment (markus-walther) [inline src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx]: `const {state:{receivedEvent}} = this;` at the beginning reads a bit cleaner.

File: src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      <div>
+        <AXAToggleSwitchReact
+          active={this.state.active}
+          onChange={this.change}
+        />
+        <p>
+          Received onchange event:
```
Comment (markus-walther) [inline src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx]: spelling: `onChange`

File: src/components/10-atoms/toggle-switch/react/demo.jsx [hunk 6]
```diff
@@ -7,6 +7,7 @@ import changelog from '../CHANGELOG.md';
 import readme from '../README.md';
 import AXAToggleSwitchReact from './AXAToggleSwitchReact';
 import ControlledWithOnChange from './ControlledWithOnChange';
+import ControlledWithDummyOnChange from './ControlledWithDummyOnChange';
```
Comment (markus-walther) [inline src/components/10-atoms/toggle-switch/react/demo.jsx]: Dummy doesn't quite describe the constant active value. No strong objection, but the point here is the constant active value, not the onChange, is it not?

File: src/components/10-atoms/toggle-switch/index.js [hunk 7]
```diff
@@ -85,9 +85,10 @@ class AXAToggleSwitch extends LitElement {
       this.active = event.target.checked;
       const { active } = this;
       fireCustomEvent('change', { active }, this);
+      this.onChange(event);
+    } else {
+      this.onChange({ target: { checked: this.active } });
```
Comment (markus-walther) [inline src/components/10-atoms/toggle-switch/index.js]: nice!

**Review by markus-walther — CHANGES_REQUESTED** [review]
Raphi, looks good.

Some small findings to think about.

Cheers, Markus

File: src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx (hunk 3 above)
Comment (raphaellueckl) [reply src/components/10-atoms/toggle-switch/react/ControlledWithDummyOnChange.jsx]: Using state for this rerenders the component differently, if `active` would not stay at the same value. Having that in `states` ensures that the `onChange` handler does not misbehave.

File: src/components/10-atoms/toggle-switch/README.md [hunk 8]
```diff
@@ -79,3 +79,13 @@ The callback function signature is `callBackFunction(nativeChangeEvent)`.
 
 A `change` event is fired on the component itself. The event bubbles and is cancellable. Its `detail` property contains
 this object: `{active}`, where the meaning of active is as defined above for the same-named property.
+
+The "onChange" handler will receive a synthetic event of the following structure:
```
Comment (markus-walther) [inline src/components/10-atoms/toggle-switch/README.md:76]: uh, the text is OK, but shouldn't this replace the "The callback function signature is `callBackFunction(nativeChangeEvent)`" bit?

File: src/components/10-atoms/toggle-switch/react/demo.jsx [hunk 9]
```diff
@@ -7,6 +7,7 @@ import changelog from '../CHANGELOG.md';
 import readme from '../README.md';
 import AXAToggleSwitchReact from './AXAToggleSwitchReact';
 import ControlledWithOnChange from './ControlledWithOnChange';
+import ControlledWithDummyOnChange from './ControlledWithConstantActiveProperty';
```
Comment (markus-walther) [inline src/components/10-atoms/toggle-switch/react/demo.jsx]: If the file has been renamed, shouldn't the imported name change, too?

**Review by markus-walther — CHANGES_REQUESTED** [review]
Raphi,

nearly there - 2 things only I found.

Cheers, Markus

**Review by markus-walther — APPROVED** [review]
LGTM

**Pull request merged**